crates/nu-command/tests/commands/for_.rs

```diff
@@ -40,3 +40,14 @@ fn failed_for_should_break_running() {
         print 3");
     assert!(!actual.out.contains('3'));
 }
+
+#[test]
+fn for_loops_dont_collect_source() {
+    let actual = nu!("
+        for i in (seq 1 10 | each { print -n $in; $in}) {
+            print -n $i
+            if $i >= 5 { break }
+        }
+    ");
+    assert_eq!(actual.out, "1122334455");
+}
```

Comment:
What a great test:
- it would never run infinite as the sequence is limited
- it ensures that we actually not collect via the `each`
- and of course it tests that we break early and don't collect too much